ansible_collections/arista/avd/docs/containers/overview.md

```diff
@@ -0,0 +1,92 @@
+
+- [Prebuilt images](https://github.com/devcontainers/images)
+- [Features](https://containers.dev/features)
+
+Dev containers can be run locally or remotely, in a private or public cloud, in a variety of [supporting tools and editors](https://containers.dev/supporting).
+
+## AVD Dev Container Variants
+
+The AVD repository provides a number of pre-build dev container images for various use cases:
+
+- **base** - image that is used to build other variants. It is not intended for direct use.
+- **dev** - an image created for AVD contributors and to support quick testing of AVD branches. The AVD collection must be mounted in this container or installed from a git branch.
+- **universal** - an image with pre-installed AVD collection and it's dependencies that can be used to build and deploy AVD configuration using a specific AVD version.
+
+All images are based on [Python Slim](https://hub.docker.com/_/python) with some additional features and extensions pre-installed.
+
+## How to Use Dev Containers
+
+Currently the only supported option is using [VSCode](https://code.visualstudio.com/) to start a dev container.
+
+To start using a dev container:
+
+- Install [VScode Dev Containers extension](https://code.visualstudio.com/docs/devcontainers/tutorial)
+- Add `.devcontainer/devcontainer.json` to your AVD repository (check examples below for possible dev container definitions).
+- Open VSCode command pallette and pick `Dev Containers: Rebuild and Reopen in Container`
+- Wait until the build will finish and enjoy running AVD in a container.
+
+If you want to use a specific AVD release, use the following `devcontainer.json`:
+
+```json
+{
+    "name": "AVD Universal",
+    "image": "ghcr.io/aristanetworks/ansible-avd/universal:python3.11-avd-v4.4.0"
+}
+```
+
+For AVD contributors the following configuration is recommended:
+
+```json
+{
+    "name": "AVD Development",
+    "image": "ghcr.io/aristanetworks/ansible-avd/dev:python3.11",
+    "mounts": [
+        "source=${localWorkspaceFolder}/ansible_collections/arista/avd,target=/home/avd/.ansible/collections/ansible_collections/arista/avd,type=bind"
+    ],
+    // run entrypoint script manually as it's ignored by dev container CLI otherwise
+    // the dev entrypoint is used to install ansible collections and requirements, as they are not included with the dev version
+    // true is required to exit "onCreateCommand" without entering ZSH
+    "onCreateCommand": "/bin/entrypoint.sh true"
+}
+```
+
+If you want to test specific AVD branch or fork without making any changes, use the following `devcontainer.json`:
+
+```json
+{
+    "name": "AVD Development",
+    "image": "ghcr.io/aristanetworks/ansible-avd/dev:python3.11",
+    "containerEnv": {
+        "AVD_GITHUB_REPO": "aristanetworks/ansible-avd",
+        "AVD_BRANCH_NAME": "devel"
```

Comment:
I guess you should mention in the doc that these must be changed to the appropriate variables, maybe use some syntax that shows where you need to edit like `<CHANGEME>`

Reply:
This is mentioned in the note below, otherwise I'd prefer a working example that can be copy-pasted. The only side effect can be that people will end up with an incorrect but working AVD version in this case. There will be more documents later explaining how to work with that.

Reply:
in that case please state explicitly in the note the parameters.

```
Do not forget to adjust the parameters like AVD version according to your use case.
```

>> this is not even from the snippet how to adjust the AVD version.